src/editor-registry.coffee

```diff
@@ -0,0 +1,35 @@
+{Emitter, Disposable} = require 'event-kit'
+
+# This global registry tracks registered `TextEditors`.
+#
+# Packages that provide extra functionality to `TextEditors`, such as
+# autocompletion, can observe this registry to find applicable editors.
+module.exports =
+class EditorRegistry
+  constructor: ->
+    @editors = []
+    @emitter = new Emitter
+
+  # Register a `TextEditor`.
+  #
+  # * `editor` The editor to register.
+  #
+  # Returns a {Disposable} on which `.dispose()` can be called to remove the
+  # added editor. To avoid any memory leaks this should be called when the
+  # editor is destroyed.
+  add: (editor) ->
+    @editors.push(editor)
+    @emitter.emit 'did-add-editor', editor
+    new Disposable =>
+      @editors = @editors.filter (e) -> e isnt editor
```

Comment:
I think this might be better implemented as a `Set` so that removal is O(1). You can still use `forEach` to iterate it when new observers are added. Any objections?